nalgebra-lapack/tests/linalg/generalized_eigenvalues.rs

```diff
@@ -1,74 +1,72 @@
-use na::dimension::{Const, Dynamic};
-use na::{DMatrix, EuclideanNorm, Norm, OMatrix};
-use nl::GE;
+use na::dimension::Const;
+use na::{DMatrix, OMatrix};
+use nl::GeneralizedEigen;
 use num_complex::Complex;
 use simba::scalar::ComplexField;
-use std::cmp;
 
 use crate::proptest::*;
-use proptest::{prop_assert, proptest};
+use proptest::{prop_assert, prop_compose, proptest};
+
+prop_compose! {
+    fn f64_dynamic_dim_squares()
+        (n in PROPTEST_MATRIX_DIM)
+        (a in matrix(PROPTEST_F64,n,n), b in matrix(PROPTEST_F64,n,n)) -> (DMatrix<f64>, DMatrix<f64>){
+    (a,b)
+}}
 
 proptest! {
     #[test]
-    fn ge(n in PROPTEST_MATRIX_DIM) {
-        let n = cmp::max(1, cmp::min(n, 10));
-        let a = DMatrix::<f64>::new_random(n, n);
-        let b = DMatrix::<f64>::new_random(n, n);
-        let a_condition_no = a.clone().try_inverse().and_then(|x| Some(EuclideanNorm.norm(&x)* EuclideanNorm.norm(&a)));
-        let b_condition_no = b.clone().try_inverse().and_then(|x| Some(EuclideanNorm.norm(&x)* EuclideanNorm.norm(&b)));
+    fn ge((a,b) in f64_dynamic_dim_squares()){
+
+        let a_c = a.clone().map(|x| Complex::new(x, 0.0));
+        let b_c = b.clone().map(|x| Complex::new(x, 0.0));
+        let n = a.shape_generic().0;
 
-        if a_condition_no.unwrap_or(200000.0) < 5.0 && b_condition_no.unwrap_or(200000.0) < 5.0 {
-            let a_c = a.clone().map(|x| Complex::new(x, 0.0));
-            let b_c = b.clone().map(|x| Complex::new(x, 0.0));
+        let ge = GeneralizedEigen::new(a.clone(), b.clone());
+        let (vsl,vsr) = ge.clone().eigenvectors();
 
-            let ge = GE::new(a.clone(), b.clone());
-            let (vsl,vsr) = ge.clone().eigenvectors();
 
-            for (i,(alpha,beta)) in ge.raw_eigenvalues().iter().enumerate() {
-                let l_a = a_c.clone() * Complex::new(*beta, 0.0);
-                let l_b = b_c.clone() * *alpha;
+        for (i,(alpha,beta)) in ge.raw_eigenvalues().iter().enumerate() {
+            let l_a = a_c.clone() * Complex::new(*beta, 0.0);
+            let l_b = b_c.clone() * *alpha;
 
-                prop_assert!(
-                    relative_eq!(
-                        ((&l_a - &l_b)*vsr.column(i)).map(|x| x.modulus()),
-                        OMatrix::zeros_generic(Dynamic::new(n), Const::<1>),
-                        epsilon = 1.0e-7));
+            prop_assert!(
+                relative_eq!(
+                    ((&l_a - &l_b)*vsr.column(i)).map(|x| x.modulus()),
+                    OMatrix::zeros_generic(n, Const::<1>),
+                    epsilon = 1.0e-5));
```

Comment:
Relative comparisons tend to break down near zero - can't this make this test relatively brittle?

Reply:
I see what you mean. I believe there is the absdiff catch as shown here https://docs.rs/approx/0.4.0/src/approx/relative_eq.rs.html#68

Reply:
I didn't know that `relative_eq!` also has an absolute tolerance check in addition - that's very unexpected.

It is a little unfortunate then that `approx` is used to liberally within `nalgebra`. Using default absolute tolerances has a number of pitfalls - in particular because it is generally impossible to know the right magnitude to use without further context.

With all that said, it might be fine in this case, if you know that the "scale" (for example, the largest entry) of your numbers are expected to be somewhat close to `1`.

Reply:
> With all that said, it might be fine in this case, if you know that the "scale" (for example, the largest entry) of your numbers are expected to be somewhat close to 1.

The eigenvectors seem to be guaranteed by LAPACK in their documentation 
> Each eigenvector is scaled so the largest component has  abs(real part)+abs(imag. part)=1.`

So I think scale wise we should be good